pages/api/teams/[slug]/members.ts

```diff
@@ -74,6 +75,10 @@ const handleDELETE = async (req: NextApiRequest, res: NextApiResponse) => {
     req.query as { memberId: string }
   );
 
+  const getMemberData = await throwIfCannotGetMember(req, memberId);
+
+  throwIfNotOwnerOnOwner(teamMember.role, getMemberData.role);
+
```

Comment:
Use the existing rbac pattern we already have.
Check the `validateUpdateRole` function